runtime/blake2.c

```diff
@@ -244,3 +244,15 @@ CAMLprim value caml_blake2_string(value hashlen, value key,
   caml_BLAKE2Final(&ctx, hlen, &Byte_u(hash, 0));
   return hash;
 }
+
+CAMLprim value caml_blake2_bytes(value hashlen, value key,
+                                  value buf, value ofs, value len)
+{
+  struct BLAKE2_context ctx;
+  size_t hlen = Long_val(hashlen);
+  caml_BLAKE2Init(&ctx, hlen, caml_string_length(key), &Byte_u(key, 0));
+  caml_BLAKE2Update(&ctx, &Byte_u(buf, Long_val(ofs)), Long_val(len));
+  value hash = caml_alloc_string(hlen);
+  caml_BLAKE2Final(&ctx, hlen, &Byte_u(hash, 0));
+  return hash;
```

Comment:
Unless I am mistaken, this is exactly the code of `caml_blake2_string` above -- there is no difference between `string` and `bytes` at the C level. Instead of copying the code unchanged, could you just call `caml_blake2_string` from `caml_blake2_bytes`?

Reply:
Done